source/extensions/network/dns_resolver/cares/dns_impl.cc

```diff
@@ -356,6 +357,25 @@ void DnsResolverImpl::AddrInfoPendingResolution::startResolution() {
 }
 
 void DnsResolverImpl::AddrInfoPendingResolution::startResolutionImpl(int family) {
+  if (filter_unroutable_families_) {
+    switch (family) {
+    case AF_INET:
+      if (!available_interfaces_.v4_available_) {
+        ENVOY_LOG_EVENT(debug, "cares_resolution_filtered", "filtered v4 lookup");
+        onAresGetAddrInfoCallback(ARES_SUCCESS, 0, nullptr);
```

Comment:
by putting the "filtering" logic here the rest of the code remain unchanged and all the PendingResolution construction, and callback logic can remain the same.

The only open thought I have here right now is if this should be considered a "success".

Reply:
No thoughts really on whether this is "success" or "failure". It probably depends on the external implications?

As an aside, I think we also need to independently filter V6 as well.